cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+                              cudaMemcpyDeviceToHost));
+    }
+
+    if (nullptr != the_column.valid) {
+      host_bitmask.resize(num_masks);
+      CUDA_RT_CALL(cudaMemcpy(host_bitmask.data(), the_column.valid,
+                              num_masks * sizeof(gdf_valid_type),
+                              cudaMemcpyDeviceToHost));
+    }
+
+    return std::make_tuple(host_data, host_bitmask);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Prints the values of the underlying gdf_column.
+   *
+   *---------------------------------------------------------------------------**/
+  void print() const {
+    // TODO Move the implementation of `print_gdf_column` here once it's removed
+    // from usage elsewhere
+    print_gdf_column(&the_column);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Functor for comparing if two elements between two gdf_columns are
+   * equal.
+   *
+   *---------------------------------------------------------------------------**/
+  struct elements_equal {
+    gdf_column lhs_col;
+    gdf_column rhs_col;
+    bool nulls_are_equivalent;
+
+    /**---------------------------------------------------------------------------*
+     * @brief Constructs functor for comparing elements between two gdf_column's
+     *
+     * @param lhs The left column for comparison
+     * @param rhs The right column for comparison
+     * @param nulls_are_equal Desired behavior for whether or not nulls are
+     * treated as equal to other nulls. Defaults to true.
+     *---------------------------------------------------------------------------**/
+    __host__ __device__ elements_equal(gdf_column lhs, gdf_column rhs,
+                                       bool nulls_are_equal = true)
+        : lhs_col{lhs}, rhs_col{rhs}, nulls_are_equivalent{nulls_are_equal} {}
+
+    __device__ bool operator()(gdf_index_type row) {
+      bool const lhs_is_valid{gdf_is_valid(lhs_col.valid, row)};
+      bool const rhs_is_valid{gdf_is_valid(rhs_col.valid, row)};
+
+      if (lhs_is_valid && rhs_is_valid) {
+        return static_cast<ColumnType const*>(lhs_col.data)[row] ==
+               static_cast<ColumnType const*>(rhs_col.data)[row];
+      }
+
+      // If one value is valid but the other is not
+      if (lhs_is_valid xor rhs_is_valid) {
+        return false;
+      }
+
+      return nulls_are_equivalent;
+    }
+  };
+
+  /**---------------------------------------------------------------------------*
+   * @brief Compares if another column_wrapper is equal to this wrapper.
+   *
+   * Treats NULL == NULL
+   *
+   * @param rhs  The other column_wrapper to check for equality
+   * @return true The two columns are equal
+   * @return false The two columns are not equal
+   *---------------------------------------------------------------------------**/
+  bool operator==(column_wrapper<ColumnType> const& rhs) const {
```

Comment:
Isn't there some existing function which compares gdf_columns?

Reply:
There was a unit testing utility function that did do this, but it was not correct because it compared the garbage held in null values: https://github.com/rapidsai/cudf/blob/branch-0.6/cpp/tests/utilities/cudf_test_utils.cuh#L336